etc/vagrant/.gitignore

```diff
@@ -0,0 +1 @@
+.vagrant
```

Comment:
I have a slight leaning towards a single `.gitignore` file - makes it easy to know what files are excluded within openthread, rather than having to search for other `.gitignore` files.